File: Week5/1122.py
class Solution:
    def relativeSortArray(self, arr1: List[int], arr2: List[int]) -> List[int]:
        dic={}
        dicre={}
        L=len(arr2)
        for i in arr2:
            dic[i]=-L
            dicre[-L]=i
            L-=1
        for index,val in enumerate(arr1):
            if val in dic:
                arr1[index]=dic[val]
        arr1.sort()
        for index,val in enumerate(arr1):
            if val<0:
                arr1[index]=dicre[val]
        return arr1

File: Week5/1122.js
var relativeSortArray = function(arr1, arr2) {
    var arr=[]
    var dic={}
    for (i of arr1){
        dic[i]=(dic[i]==undefined?0:dic[i])+1
    }
    cnt=0
    for (j of arr2){
        if(j in dic){
            while(dic[j]!=0){
                arr[cnt++]=j
                dic[j]--
            }
            delete dic[j]
        }
    }
    for (let i in dic){
        while(dic[i]!=0){
            arr[cnt++]=i.valueOf()
            dic[i]--
        }
    }
    return arr
};